src/utils/error-handling.ts

```diff
@@ -0,0 +1,18 @@
+import { notifications } from '../components/notifications/notifications';
+import { go, GO_ERROR_INDEX, GO_RESULT_INDEX, isGoSuccess, isUserRejection } from './generic';
+import { messages } from './messages';
+
+export const handleTransactionError = async <T>(transaction: Promise<T>) => {
```

Comment:
I like wrapping all transactions with a single helper like this but I find the name a misleading. How about something like `handleTransactionSafe` or something like that?

Reply:
I think the name is good enough. It handles transaction error (thus the reasoning for this name).